### ISSUE TITLE
💅 noImportCycles reports type improts

### ISSUE THREAD

**Issue by ziimakc** [biomejs/biome#6533]
### Environment information

```bash
-
```

### Rule name

noImportCycles

### Playground link

- doesn't support multiple files

### Expected result

Type import should not produce error.

```
// a.ts
import type { b } from "./b";
```

```
// b.ts
import { a } from "./a";
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
It's not a bug, but there is a request to add this as a feature: https://github.com/biomejs/biome/discussions/6034
